src/ai/defaultai.cc

```diff
@@ -3350,6 +3352,35 @@ void DefaultAI::diplomacy_actions(const Time& gametime) {
                                           (accept ? Widelands::DiplomacyAction::kAcceptJoin :
                                                     Widelands::DiplomacyAction::kRefuseJoin)),
 			                             pda.sender);
+			verb_log_dbg_time(
+			   gametime, "Player(%d), %s the invitation of player (%d) with diploscore: %d\n",
+			   static_cast<unsigned int>(pda.other), accept ? "accepts" : "denies",
+			   static_cast<unsigned int>(pda.sender), player_statistics.get_diplo_score(pda.sender));
+		}
+	}
+	for (Widelands::PlayerNumber opn = 1; opn <= game().map().get_nrplayers(); ++opn) {
+		const Widelands::Player* other_player = game().get_player(opn);
+		// other player needs to exist and different from us
+		// we need to be still alive and we don't have send a request in last 5 + X minutes
+		if (other_player == nullptr || opn == mypn || me->is_defeated() ||
+		    player_statistics.player_diplo_requested_lately(opn, gametime)) {
+			continue;
+		}
+		if (player_statistics.get_diplo_score(opn) >= 30)
+			&&other_player->team_number() != me->team_number() {
```

Comment:
There's a compile error here.

```suggestion
		if (player_statistics.get_diplo_score(opn) >= 30 &&
		    other_player->team_number() != me->team_number()) {
```